src/toast/tests/template_hwpss.py

```diff
@@ -0,0 +1,956 @@
+            hwp_rpm=1.0,
+            obs_per_group=10,
+            pixel_per_process=7,
+            single_group=True,
+            width=5.0 * u.degree,
+        )
+
+        # Create detector data for the combined input signal
+        for ob in data.obs:
+            ob.detdata.create("input", units=defaults.det_data_units)
+
+        # Create a default noise model
+        default_model = ops.DefaultNoiseModel()
+        default_model.apply(data)
+
+        # Generate a fake "sky" and scan into timestreams.  We
+        # create a temporary pointing matrix and then clean up
+        # temporary data objects afterwards.
+
+        detpointing = ops.PointingDetectorSimple(
+            boresight=defaults.boresight_radec,
+            quats="temp_quats",
+        )
+        detpointing.apply(data)
+
+        pixels = ops.PixelsHealpix(
+            nside=sky_nside,
+            detector_pointing=detpointing,
+            pixels="temp_pix",
+        )
+        pix_dist = ops.BuildPixelDistribution(
+            pixel_dist="sky_pixel_dist",
+            pixel_pointing=pixels,
+        )
+        pix_dist.apply(data)
+
+        create_fake_sky(data, "sky_pixel_dist", "fake_map")
+
+        outfile = os.path.join(outdir, f"sky_{self.nside}_input.fits")
+        write_healpix_fits(data["fake_map"], outfile, nest=True)
+        if data.comm.world_rank == 0:
+            plot_healpix_maps(mapfile=outfile)
+
+        weights = ops.StokesWeights(
+            mode="IQU",
+            hwp_angle=defaults.hwp_angle,
+            weights="temp_weights",
+            detector_pointing=detpointing,
+        )
+
+        scanner = ops.Pipeline(
+            operators=[
+                pixels,
+                weights,
+                ops.ScanMap(
+                    det_data="sky",
+                    pixels=pixels.pixels,
+                    weights=weights.weights,
+                    map_key="fake_map",
+                    det_data_units=defaults.det_data_units,
+                ),
+            ],
+            detsets=["SINGLE"],
+        )
+        scanner.apply(data)
+
+        ops.Combine(
+            op="add",
+            first="sky",
+            second="input",
+            result="input",
+        ).apply(data)
+
+        del data["fake_map"]
+        del data["sky_pixel_dist"]
+        for ob in data.obs:
+            del ob.detdata["temp_weights"]
+            del ob.detdata["temp_pix"]
+            del ob.detdata["temp_quats"]
+
+        # Simulate some noise
+        sim_noise = ops.SimNoise(
+            det_data="noise",
+            noise_model=default_model.noise_model,
+            det_data_units=defaults.det_data_units,
+        )
+        sim_noise.apply(data)
+        ops.Combine(
+            op="add",
+            first="noise",
+            second="input",
+            result="input",
+        ).apply(data)
+
+        return data
+
+    def create_ground_sim(self, outdir, width, sky_proj, sky_res):
```

Comment:
Don't we already have helper functions for this?

Reply:
We have helper functions to simulate a small observing schedule (which is called by this function), but we do not have a higher level function that simulates all the components (since that might be different between tests).  We could extract this in the future though if we want to use it in multiple tests.